README.md

```diff
@@ -198,54 +213,204 @@ The method `evaluate_quadratic_expression` has the following interface:
+### Predefined Unsigned Integers
 
-BigNumParams parameters can be derived from a known modulus using the rust crate `noir-bignum-paramgen` (https://crates.io/crates/noir-bignum-paramgen)
+BigNum supports operations over unsigned integers, with predefined types for 256, 384, 512, 768, 1024, 2048, 4096 and 8192 bit integers. Keep in mind these are not Fields.
 
-## Additional usage examples
+All arithmetic operations are supported including integer div and mod functions (make sure to use udiv, umod). Bit shifts and comparison operators are not yet implemented.
+
+```rust
+use dep::bignum::fields::U256::U256Params;
+use dep::bignum::BigNum;
+
+type U256 = BigNum<3, 257, U256Params>;
+
+fn foo(x: U256, y: U256) -> U256 {
+    x.udiv(y)
+}
+```
+
+### Predefined Fields
+
+This library includes the follow field presets:
+
+| Curve       | Base Field Instance                | Scalar Field Instance              |
+|-------------|------------------------------------|------------------------------------|
+| [BLS12-377](https://eprint.iacr.org/2020/351)   | `BLS12_377_Fq_Params`            | `BLS12_377_Fr_Params`            |
+| [BLS12-381](https://electriccoin.co/blog/new-snark-curve/)   | `BLS12_381_Fq_Params`            | `BLS12_381_Fr_Params`            |
+| [BN254](https://zips.z.cash/protocol/protocol.pdf)   | `BN254_Fq_Params`                    | (Scalar field is the native field in Noir) |
+| [Curve25519](https://cr.yp.to/ecdh/curve25519-20060209.pdf)  | `ED25519_Fq_Params`              | `ED25519_Fr_Params`              |
+| [MNT4-753](https://eprint.iacr.org/2014/595)    | `MNT4_753_Fq_Params`             | `MNT4_753_Fr_Params`             |
+| [MNT6-753](https://eprint.iacr.org/2014/595)    | `MNT6_753_Fq_Params`             | `MNT6_753_Fr_Params`             |
+| [Pallas](https://electriccoin.co/blog/the-pasta-curves-for-halo-2-and-beyond/)*      | `Pallas_Fr_Params` | `Pallas_Fq_Params` |
+| [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) | `Secp256k1_Fq_Params`            | `Secp256k1_Fr_Params`            |
+| [Secp256r1](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) | `Secp256r1_Fq_Params`            | `Secp256r1_Fr_Params`            |
+| [Secp384r1](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) | `Secp384r1_Fq_Params`            | `Secp384r1_Fr_Params`            |
+
+* TODO: this should probably be adjusted, to avoid confusion
+
+Feature requests and/or pull requests welcome for missing fields you need.
+
+
+### Create a new parameter set for custom modulus
+
+TODO: the paramgen tool is not up to date for `BigNum` version >= `0.4`, an issue has been created for this [here](https://github.com/noir-lang/noir-bignum-paramgen/issues/4). This section should be adjusted after this fix.
+
+The easiest way to generate everything you need for a parameter set is to use [this tool](https://github.com/noir-lang/noir-bignum-paramgen). 
```

Comment:
In most places I've seen "parameterset" as one word, is that intentional? In a couple of places it is "parameter set".